### PR TITLE
43 Deployment

### PR DESCRIPTION
**Pull request by lojzatran**
Fixes: https://github.com/commercetools/commercetools-adyen-integration/issues/43

### CODE REVIEW

File: extension/k8s/values.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  annotations:
+    kubernetes.io/ingress.class: 'gce'
+    kubernetes.io/ingress.allow-http: 'false'
+
+livenessProbe:
+  httpGet:
+    path: ''
+
+
+readinessProbe:
```
Comment (butenkor) [inline extension/k8s/values.yaml]: Do we have a TODO for that?

File: notification/k8s/values.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+# When the pod is scheduled, the container is guaranteed the amount of resources requested 2GiB RAM and 300m CPU time and it is not allowed to exceed 2GB RAM and 300m of CPU time.
+# This classifies the pod as "Guaranteed"
+#resources:
+#  requests:
+#    cpu: '300m' #30% of CPU time on 1 GCP Core
+#    memory: '1Gi' #equivalent to 2GiB
+#  limits:
+#    cpu: '300m' #30% of CPU time on 1 GCP Core
+#    memory: '1Gi' #equivalent to 2GB
```
Comment (butenkor) [inline notification/k8s/values.yaml]: Why no resources requests/limits are set?

File: notification/resources/payment-interface-interaction-type.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  ],
+  "fieldDefinitions": [
+    {
+      "name": "timestamp",
+      "label": {
+        "en": "timestamp"
+      },
+      "required": true,
+      "type": {
+        "name": "String"
```
Comment (butenkor) [inline notification/resources/payment-interface-interaction-type.json]: There is also date/time type. Is String on purpose here?

File: notification/src/config/init/ensure-interface-interaction-custom-type.js [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
-        await ctpClient.create(ctpClient.builder.types, type)
-    } catch (e) {
-      console.error('Error when creating interface interaction custom type, skipping...', JSON.stringify(e))
+  try {
+    logger.debug('Ensuring interfaceInteraction')
+    const { body } = await ctpClient.fetch(ctpClient.builder.types.where(`key="${interfaceInteractionType.key}"`))
+    if (body.results.length === 0) {
+      logger.debug('Creating interface interaction')
+      await ctpClient.create(ctpClient.builder.types, interfaceInteractionType)
+      logger.info('Successfully created an interfaceInteraction type')
```
Comment (butenkor) [inline notification/src/config/init/ensure-interface-interaction-custom-type.js]: I would output the key of the type too

File: notification/k8s/values.yaml (hunk 2 above)
Comment (LEQADA) [reply notification/k8s/values.yaml]: We are in a faze of figuring it out 🙂

File: notification/src/handler/notification/notification.handler.js [hunk 5]
```diff
@@ -37,8 +37,10 @@ async function updatePaymentWithRepeater (payment, notification, ctpClient) {
     try {
       /* eslint-disable-next-line no-await-in-loop */
       await ctpClient.update(ctpClient.builder.payments, currentPayment.id, currentVersion, updateActions)
+      logger.debug(`Payment with interfaceId ${currentPayment.interfaceId}` +
+        'was successfully updated')
       break
-    } catch (err) {
+    } catch (e) {
```
[the remark this answers is not in the document]
Comment (LEQADA) [reply notification/src/handler/notification/notification.handler.js]: there was a test and it was failing. Fixed the issue. Thanks

File: extension/k8s/values.yaml (hunk 1 above)
Comment (LEQADA) [reply extension/k8s/values.yaml]: https://github.com/commercetools/commercetools-adyen-integration/issues/50

File: extension/src/config/init/ensure-interface-interaction-custom-type.js [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
     try {
       const { body } = await ctpClient.fetch(ctpClient.builder.types.where(`key="${type.key}"`))
-      if (body.results.length === 0)
+      if (body.results.length === 0) {
         await ctpClient.create(ctpClient.builder.types, type)
+        logger.info('Successfully created an interfaceInteraction type')
+      }
     } catch (e) {
-      console.error('Error when creating interface interaction custom type, skipping...', JSON.stringify(e))
+      logger.error('Error when creating interface interaction custom type, skipping...', JSON.stringify(e))
```
[the remark this answers is not in the document]
Comment (LEQADA) [reply extension/src/config/init/ensure-interface-interaction-custom-type.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/44/commits/7e1038764e09cdb945f1db88b9abd0b677cc09d1

File: notification/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,13 @@
+FROM mhart/alpine-node:10
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply notification/Dockerfile]: node 10 is now in beta: https://cloud.google.com/functions/docs/concepts/nodejs-10-runtime

File: extension/src/config/init/ensure-payment-custom-type.js [hunk 8]
```diff
@@ -1,12 +1,18 @@
...
+
+const logger = utils.getLogger()
+
 async function ensurePaymentCustomType (ctpClient) {
   try {
     const { body } = await ctpClient.fetch(ctpClient.builder.types.where(`key="${paymentCustomType.key}"`))
-    if (body.results.length === 0)
+    if (body.results.length === 0) {
       await ctpClient.create(ctpClient.builder.types, paymentCustomType)
+      logger.info('Successfully created payment custom type')
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply extension/src/config/init/ensure-payment-custom-type.js]: Right now we don't have multitenancy support, so it doesn't have tenant info.

File: notification/src/config/init/ensure-interface-interaction-custom-type.js (hunk 4 above)
Comment (lojzatran) [reply notification/src/config/init/ensure-interface-interaction-custom-type.js]: As stated above, I would change it in a different task: https://github.com/commercetools/commercetools-adyen-integration/issues/8

File: notification/resources/payment-interface-interaction-type.json (hunk 3 above)
Comment (lojzatran) [reply notification/resources/payment-interface-interaction-type.json]: https://github.com/commercetools/commercetools-adyen-integration/pull/44/commits/64192281d8717eb9cf7b7097794a3cdbe6c0038a

**Pull request merged**